docs/pipeline.md

```diff
@@ -0,0 +1,177 @@
+
+## Side Streams
+
+### SS 1: External Resource Whitelister
+
+[[Code]](../wikidatarefisland/external_identifiers/generate_whitelisted_ext_ids.py), [[Makefile Command]](../Makefile#L7-L8): `make data/whitelisted_ext_idefs.json`
+
+This service produces a list of External Identifier resource which are viable candidates for data extraction. It test-scrapes a sample of 10 use cases from each non [***blacklisted external identifier***](../config/default.yml#L78-L106) to determine whether the resource of that identifier contains enough viable data to scrape. It then collects all identifiers representing viable resources and writes their Wikibase Property IDs to a JSON array.
+
+###  SS 2: Schema.org JSON-LD context fetcher
+
+[[Code]](../wikidatarefisland/data_access/schema_context_downloader.py), [[Makefile Command]](../Makefile#L12-L13): `make data/schema_org_context.jsonld`
+
+This service downloads the Schema.org JSON-LD context to prevent multiple calls to http://schema.org during scraping.
+
+Until 2020-05-19 it was possible for PyLD to automatically obtain it through content-negotiation of schema.org but this broke. To mitigate this, the side-stream has a backup method to get the context from the schema.org docs.
+
+## Noteworthy Utility Classes
+
+### Wikidata External Id URL Formatter
+
+[[Code]](../wikidatarefisland/services/external_identifier_formatter.py)
+
+This service takes in a string representation of an external id property and attempts to output a formatted URL  for an external resource, as well as reference metadata according to the Wikidata mapping. See [`ResourceBlob`](result.md#resourceblob) for output. If no formatter is found the service returns `false`.
+
+### Schema.org Data Normalizer
+
+[[Code]](../wikidatarefisland/data_model/schemaorg_normalizer.py)
+
+This service takes in an object containing raw scraped data in `json-ld` format (For example, the results of the following scraping library: https://github.com/scrapinghub/extruct with the [`uniform` option set to true](https://github.com/scrapinghub/extruct#uniform)).
+
+The output of this service will be a list of objects representing a Schema.org type with the following structure: 
+
+```js
+[
+    {
+        "http://schema.org/name": [ "Ludwig Wittgenstein" ],
+        "http://schema.org/sameAs": [ "http://viaf.org/viaf/24609378" ],
+        "http://schema.org/birthPlace": {
+          "http://schema.org/name": [ "Vienna" ],
+          "http://schema.org/geo": {
+            "http://schema.org/geo/latitude": "48.20849",
+            "http://schema.org/geo/longitude": "16.37208"
+          }
+        }
+    },
+    //...
+]
+```
+
+### Wikidata - Schema.org Property Mapper
+
+[[Code]](../wikidatarefisland/services/schemaorg_property_mapper.py)
+
+A service to retrieve the most recent state of mappings between Wikidata Properties and Schema.org properties.
```

Comment:
Perhaps worth mentioning it does so using SPARQL?

Reply:
Will do, also for SS1, and potentially the formatter (which use WDQS)